### ISSUE TITLE
Lubuntu 20.04.3 64-bit, Ketarin crashes

### ISSUE THREAD

**Issue by JimH44** [franc6/Ketarin#59]
Hi Tom,
I have moved our server to Lubuntu 20.04.3, 64-bit, to be more up-to-date and to rearrange the disk space allocation at the same time.
On the new system I did
    
    sudo apt-get update
    sudo apt-get install mono-complete

I have downloaded the zip files for 1.0.46 and 1.0.47 and unpacked them, and both fail with essentially the same error messages, as in the attached files:

[StartKetarin-46.log](https://github.com/franc6/Ketarin/files/7201645/StartKetarin-46.log)
[StartKetarin-47.log](https://github.com/franc6/Ketarin/files/7201646/StartKetarin-47.log)

I'm wondering if I need to get the 64-bit version, or if there is more I need to do in installing Mono.

Thanks for thinking about this,
Jim

**Comment by franc6** [comment]
Hi, Jim!

I’m sorry, but I didn’t get a chance to start looking at this until almost 9pm my time.  As I started, I realized I’m not sure what your upgrade process was.  Can you tell me what you did to upgrade Lubuntu?  I’d just like to do it the same way.  It might take me a couple of days, just because I’m more limited on time right now. :(

Your upgrade for mono looks good, but it seems to be picky sometimes. :)  I’m praying to see the same problem, and that maybe the fix will be simple. 

TOM

-- 
Preserve wildlife -- pickle a squirrel!

**Comment by JimH44** [comment]
Sorry, Tom, I didn't notice your email.

I asked backupsy.com to make me a new 64-bit server and mount the 
installer for Lubuntu 20.04 on it. I then installed the operating system 
with the partitioning that I wanted.

I then used rsync to copy the Langtran-specific files from the old 
server to folders on the new computer and used my "update" script to 
copy things to the right places on the new server without replacing 
anything newer.

Then I asked Backupsy to remove the old server and give the new server 
the same name and IP address as the old server had had.

I don't think there is a way to upgrade Lubuntu from 32-bit to 64-bit in 
place. I think it requires a new installation.

Thanks again,
Jim

**Comment by franc6** [comment]
I've been able to reproduce the problem, and unfortunately, it's not something I can fix.  I did make one update for x64 builds, as I couldn't run the latest build due to DLL problems.  But after that, I get the errors you reported.  The problem is that the graphics environment is returning a size of 0 for the cursor, and when mono attempts to use that size, it crashes.  There are some older reports of the same issue on other platforms, so it's rather surprising to me that this bug appears in a more recent linux distribution.  No changes were made to mono since it's not considered a bug in mono (even though a work-around could be applied to mono).

I can confirm that if I run Ketarin to be displayed on a system without the bug (e.g. macos or freebsd), it works just fine.  If you have another system running X11, you might be able to run it as I did.  In the meantime, there could be an update for Lubuntu's X11 packages or to mono which will eventually resolve the problem (at least one of the bug reports is still "open" so maybe mono will add a work-around).

I'm going to check ubuntu 21.04 to see if it's also problematic, I can confirm that 20.04 is a problem, too.
